### ISSUE TITLE
Hostname should support Internationalized domain name (IDN) TLDs

### ISSUE THREAD

**Issue by candita** [go-openapi/strfmt#94]
Hostname RegEx should support a TLD that exists in the IANA list of TLDs at http://data.iana.org/TLD/tlds-alpha-by-domain.txt,
e.g. XN--11B4C3D, or कॉम.  

Currently it disallows numbers and hyphen in the last label of the Hostname (TLD):
`^([a-zA-Z0-9\p{S}\p{L}]((-?[a-zA-Z0-9\p{S}\p{L}]{0,62})?)|([a-zA-Z0-9\p{S}\p{L}](([a-zA-Z0-9-\p{S}\p{L}]{0,61}[a-zA-Z0-9\p{S}\p{L}])?)(\.)){1,}([a-zA-Z\p{L}]){2,63})$
`

Can I change it to allow digits and hyphens in the TLD, or create a new format called HostnameIDN?
`^([a-zA-Z0-9\p{S}\p{L}]((-?[a-zA-Z0-9\p{S}\p{L}]{0,62})?)|([a-zA-Z0-9\p{S}\p{L}](([a-zA-Z0-9-\p{S}\p{L}]{0,61}[a-zA-Z0-9\p{S}\p{L}])?)(\.)){1,}([a-zA-Z0-9-\p{L}]){2,63})$
`

**Comment by casualjim** [comment]
Yeah, but maybe we should just go with https://pkg.go.dev/golang.org/x/net/idna ?

**Comment by candita** [comment]
You would make the Hostname RegEx obsolete?

**Comment by casualjim** [comment]
Yeah, we did the same for email addresses because regex'es are always a bit wonky for these things.
https://github.com/go-openapi/strfmt/blob/master/default.go#L122-L126

**Comment by fredbi** [comment]
@candita  I've pushed a PR that goes the regexp way, but with a very localized use of the idna package.
The problem with the idna package is that it covers only partially the set of validation rules.

Also I've hit an issue with the devanagari example that you supplied: these rune do not classify as unicode letters, so my PR doesn't cover these. I am not aware of any regexp trick to easily cover such runes.

Cheers

Fred

**Comment by fredbi** [comment]
New PR now addresses the above concerns
